src/util/SO3.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+import numpy as np
+from tf.transformations import (quaternion_inverse, quaternion_matrix, quaternion_from_matrix,
+                                quaternion_multiply, rotation_from_matrix)
+
+
+class SO3:
+    quaternion: np.ndarray
+
+    def __init__(self, quaternion: np.ndarray = None):
```

Comment:
for stuff to work this has to be a unit quaternion. We should either note that in the docstring or forcibly normalize the input